docs/src/lens.md

```diff
@@ -1,149 +1,251 @@
+
+> Side note: Doing this makes using the lenses very simple (you just do `StructName::field_name`), but it can be a bit confusing, because of breaking the naming conventions. This is the reason I've included the expanded code in the page.
 
-The easiest way to use a lens is with the `lens` method that is provided through
-the `WigetExt` trait; this is a convenient way to wrap a widget in a `LensWrap`
-with a given lens.
+### Composing lenses
+
+If I told you that the concept of lenses comes from Haskell (the functional megolith), I'm sure you won't be suprised when I also tell you that they really excel when it comes to composition. Let's say we have an outer struct that contains an inner struct, with the inner struct containing a `String`. Now let's say we want to tell a label widget to display the string as text in a label. We could write a lens from the outer struct to the string, which would look something like `f(&outer.inner.text)`, but actually we don't need to do this: we can use the `then` combinator. The full example is below
+
+```rust
+#[derive(Lens)]
+struct Outer {
+    inner: Inner,
+}
 
-Let's build the UI for our todo list item:
+#[derive(Lens)]
+struct Inner {
+    text: String
+}
 
-```rust,noplaypen
-{{#include ../book_examples/src/lens_md.rs:build_ui}}
+// `composed_lens` will contain a lens that goes from `Outer` through `Inner` to `text`.
+let composed_lens = Outer::inner.then(Inner::text);
 ```
 
-## Advanced lenses
+> Side note: Because unlike Haskell, Rust has all the type information during monomorphisation, it can inline all these functions and the extra cost of having 2 lens functions disappears, leaving what you would have written by hand. When you're waiting for Rust's infamously long compilations to finish, know that you're waiting for a potentially significant run-time benefit in exchange.
 
-Field access is a very simple (and common, and *useful*) case, but lenses can do much more than that.
+`LensExt` contains a few more useful methods for handling things like negating a boolean, or auto-`Deref`ing a value.
 
-### `LensExt` and combinators
+There are also 3 special structs in `druid::lens`: `Constant`, `Identity` and `Unit`. `Constant` is a lens that always returns the same value, and always discards any changes, while `Identity` is a lens that does nothing. You might say "what is the point of a lens that does nothing", which would be a fair question. Well, there are some places where a lens is required, and having an identity allows the user to say act as if there was no lens. It's also used to begin a composition chain using the combinators like `then`. `Unit` is a special case of `Constant` where the constant in question is `()`.
```

Comment:
I'm currently trying to understand how the lens works, and that's typically the sort of thing I'm looking to clear up: When to use Identity ? some idiomatic examples would be welcome. There is a case in "examples/list.rs" but it is not obvious to me as a novice.